File: xor/cpp/xor.cpp
#include <iostream>
#include <fstream>
#include <ctype.h>
#include <string>
#include <cstring>
#include <iterator>
using namespace std;

string xor_cipher(string text, string password) {
	string cipher_text = "";
	for (int i = 0; i < text.length(); i++) {
		cipher_text += text[i] ^ password[i % password.length()];
	}
	return cipher_text;
}

int main(int argc, char *argv[]) {
	string password;
	string in_file;
	string out_file;
	if (argc == 4) {
		password = argv[1];
		in_file = argv[2];
		out_file = argv[3];
	}
	else return 0;

	ifstream ifs(in_file, ios::in | ios::binary);
	string text((istreambuf_iterator<char>(ifs)),
				(istreambuf_iterator<char>()));
	ifs.close();
	string cipher_text = xor_cipher(text, password);
	ofstream encrypted(out_file, ios::out | ios::binary);
	encrypted.write(cipher_text.c_str(), cipher_text.length());
	encrypted.close();
}

File: xor/python/xor.py
import argparse

def xor(text, password):
	cipher_text = bytearray()
	for i in range(len(text)):
		cipher_text.extend((int.from_bytes(text[i:i+1], byteorder='big') ^ password[i % len(password)]).to_bytes(1, byteorder='big'))
	return cipher_text

def main():
	parser = argparse.ArgumentParser()
	parser.add_argument("--password")
	parser.add_argument("--in_file")
	parser.add_argument("--out_file")
	args = parser.parse_args()

	file = open(args.in_file, "rb")
	text = file.read()
	file.close()

	e_text = open(args.out_file, 'wb')
	e_text.write(xor(text, args.password.encode()))
	e_text.close()

main()
